src/EventListener/RenderingForwarder.php

```diff
@@ -70,28 +69,35 @@ public function registerTemplates(): void
     /**
      * @Hook("parseTemplate")
      */
-    public function storeTemplateContext(Template $template): void
+    public function delegateRendering(Template $contaoTemplate): void
     {
-        $this->templateContext[$template->getName()] = $template->getData();
+        $template = $this->templates[$contaoTemplate->getName()] ?? null;
+
+        if (null === $template) {
+            return;
+        }
+
+        // delegate to our proxy template that will call render()
+        $contaoTemplate->setName('twig_template_proxy');
+        $contaoTemplate->{self::TWIG_TEMPLATE_KEY} = $template;
```

Comment:
I'd prefer keeping old data untouched, for example:

```php
$contaoTemplate->setData(
  self::TWIG_TEMPLATE_KEY = $template,
  self::TWIG_TEMPLATE_CONTEXT = $contaoTemplate->getData()
);
```

Reply:
Good idea.